src/components/organisms/Modal/Modal.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  FC,
+  ReactNode,
+  PropsWithChildren,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import classNames from 'classnames'
+import Button, { AppearanceTypes } from 'components/molecules/Button/Button'
+import { ReactComponent as Close } from 'assets/icons/close.svg'
+import { useTranslation } from 'react-i18next'
+
+import classes from './styles.module.scss'
+
+export enum ModalSizeTypes {
+  Narrow = 'narrow',
+  Medium = 'medium',
+  Big = 'big',
+}
+
+export enum ButtonPositionTypes {
+  SpaceBetween = 'spaceBetween',
+  Right = 'right',
+  None = 'none',
+}
+
+export enum TitleFontTypes {
+  Gray = 'gray',
+  Black = 'black',
+}
+
+export interface ModalProps {
+  title?: string
+  trigger?: ReactNode
+  customDialogContent?: string
+  topButton?: boolean
+  size?: ModalSizeTypes
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
```

Comment:
**[non-blocker]** (for now at least)

I would suggest leaving this type a bit more open. Currently our modal can only be used, if we have some react setState function to pass to setOpen, however we are currently not 100% if we will have this every time. We might use some other state management for holding the state of modal in some rare cases.

Reply:
Ok, got it, I'll change it then when we have more information